recipes/qcustomplot/all/conanfile.py

```diff
@@ -39,17 +40,14 @@ def config_options(self):
     def configure(self):
         if self.options.shared:
             self.options.rm_safe("fPIC")
-        # FIXME: we shouldn't have to force shared in qt, but config file
-        #        generated by conan in qt static is likely broken, or maybe env vars.
-        self.options["qt"].shared = True
+        self.options["qt"].gui = True
+        self.options["qt"].widgets = True
 
     def requirements(self):
         if Version(self.version) >= "2.0.0":
-            self.requires("qt/6.4.1")
+            self.requires("qt/[>=6.4 <7]", transitive_headers=True, transitive_libs=True)
```

Comment:
So this never worked in Conan 2 then? Nice catch if so!

Reply:
Why transitive_libs BTW?

Reply:
test_package without transitive_libs=True:
```
undefined reference to symbol '_ZN7QWidget11setGeometryERK5QRect@@Qt_6'
qt00e75f421b97e/p/lib/libQt6Widgets.so.6: error adding symbols: DSO missing from command line
```